cli/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala

```diff
@@ -0,0 +1,103 @@
+package com.sourcegraph.lsif_java.buildtools
+
+import java.nio.charset.StandardCharsets
+import java.nio.file._
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Properties
+
+import com.sourcegraph.lsif_java.IndexCommand
+import os.CommandResult
+
+class GradleBuildTool(index: IndexCommand) extends BuildTool("Gradle", index) {
+
+  override def defaultTargetroot: Path =
+    Paths.get("build", "semanticdb-targetroot")
+
+  override def usedInCurrentDirectory(): Boolean = {
+    Files.isRegularFile(index.workingDirectory.resolve("settings.gradle")) ||
+    Files.isRegularFile(index.workingDirectory.resolve("gradlew")) ||
+    Files.isRegularFile(index.workingDirectory.resolve("build.gradle")) ||
+    Files.isRegularFile(index.workingDirectory.resolve("build.gradle.kts"))
+  }
+
+  override def generateSemanticdb(): CommandResult = {
+    val gradleWrapper: Path = index
+      .workingDirectory
+      .resolve(
+        if (Properties.isWin)
+          "gradlew.bat"
+        else
+          "gradlew"
+      )
+    val gradleCommand: String =
+      if (Files.isRegularFile(gradleWrapper))
+        gradleWrapper.toString
+      else
+        "gradle"
+
+    TemporaryFiles.withDirectory(index.cleanup) { tmp =>
+      val toolchains = GradleJavaToolchains
+        .fromWorkspace(this, index, gradleCommand, tmp)
+      val script = initScript(toolchains, tmp).toString
+
+      val buildCommand = ListBuffer.empty[String]
+      buildCommand ++=
+        List(
+          gradleCommand,
+          "--init-script",
+          script,
+          "-Porg.gradle.java.installations.auto-detect=false",
+          "-Porg.gradle.java.installations.auto-download=false",
+          s"-Porg.gradle.java.installations.paths=${toolchains.paths()}",
+          s"--no-daemon"
+        )
+      buildCommand ++= index.finalBuildCommand(List("clean", "compileTestJava"))
```

Comment:
Curious what this is for, why `compileTestJava` is hardcoded

Reply:
Is there a better default command for gradle? I tried `-x test build` but it often skips compilation of test files and it also triggers integration tests. The user can override this command like this
```
lsif-java -- my custom gradle command
```

Reply:
Ah okay didn't know its overridable, seems fine then :slightly_smiling_face: Ive a feeling some of the android projects arent emitting semanticdb files due to using different compile task names, but Im not sure, would've thought `build` would catch it

Reply:
The injected gradle settings specifically target the java plugin, which isn't used in android projects. I saw that the errorprone plugin had android specific logic, which we may need to learn from